src/vs/workbench/contrib/search/browser/searchWidget.ts

```diff
@@ -326,7 +356,14 @@ export class SearchWidget extends Widget {
 		};
 
 		const searchInputContainer = dom.append(parent, dom.$('.search-container.input-box'));
-		this.searchInput = this._register(new ContextScopedFindInput(searchInputContainer, this.contextViewService, inputOptions, this.contextKeyService));
+
+		const experimentalNotebooksEnabled = this.configurationService.getValue<ISearchConfigurationProperties>('search').experimental.notebookSearch;
+		if (experimentalNotebooksEnabled) {
+			const scopedContextKeyService = this._register(this.contextKeyService.createScoped(searchInputContainer));
+			this.searchInput = this._register(new NotebookFindInput(this._filters, scopedContextKeyService, this.contextMenuService, this.instantiationService, searchInputContainer, this.contextViewService, inputOptions, this._hasNotebookOpen()));
```

Comment:
It actually seem a bit odd to swap out the whole find input rather than just change which options are available. I don't think we should import a whole widget from the notebook side of things into search, especially if you have to make changes to it to work the way that search wants it to work.